redux/middleware/firebase.js

```diff
@@ -0,0 +1,23 @@
+import firebaseApp from '~/firebase/index';
+
+export default store => next => action => {
+    const { type, payload } = action;
+
+    // If firebase action, update firebase
+    if ( type.match(/^FIREBASE_/) ) {
```

Comment:
Middleware to handle firebase actions.

We should firebase actions as normal, but if we want them to be processed by firebase, make sure they are prefixed with FIREBASE_.

In this case, all FIREBASE_ actions just get swallowed here, and direct communication with firebase happens here.